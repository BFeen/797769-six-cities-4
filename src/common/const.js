const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorite`,
  DETAILS: `/offer`,
};

const Cities = [
  {
    name: `Paris`,
    coordinates: [48.85, 2.34],
    zoom: 13,
  }, {
    name: `Cologne`,
    coordinates: [50.93, 6.95],
    zoom: 13,
  }, {
    name: `Brussels`,
    coordinates: [50.84, 4.35],
    zoom: 13,
  }, {
    name: `Amsterdam`,
    coordinates: [52.38333, 4.9],
    zoom: 13,
  }, {
    name: `Hamburg`,
    coordinates: [53.55, 9.99],
    zoom: 13,
  }, {
    name: `Dusseldorf`,
    coordinates: [51.22, 6.77],
    zoom: 13,
  }
];

const ClassNames = {
  MapClassNames: {
    CITIES: `cities`,
    PROPERTY: `property`,
  },
  LoginClassNames: {
    AUTH: `header__user-name user__name`,
    NO_AUTH: `header__login`,
  },
  PlacesListClassNames: {
    MAIN: `cities__places-list tabs__content`,
    DETAILS: `near-places__list`,
    FAVORITES: `favorites__places`,
  },
  CardClassNames: {
    MAIN: `cities__place-card`,
    DETAILS: `near-places__card`,
    FAVORITES: `favorites__card`,
  },
  ButtonClassNames: {
    SIGN_IN: `login__submit form__submit button`,
    REVIEW: `reviews__submit form__submit button`,
  },
};

const ScreenType = {
  MAIN: `MAIN`,
  DETAILS: `DETAILS`,
  FAVORITES: `FAVORITES`,
};

const MONTHS = [
  `January`,
  `February`,
  `March`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `October`,
  `November`,
  `December`,
];

const RatingStarsCount = {
  FIVE: `perfect`,
  FOUR: `good`,
  THREE: `not bad`,
  TWO: `badly`,
  ONE: `terribly`,
};

const SortType = {
  POPULAR: `Popular`,
  LOW_TO_HIGH: `Price: low to high`,
  HIGH_TO_LOW: `Price: high to low`,
  RATING: `Top rated first`,
};

const ServerMessage = {
  [`400`]: `Server is not available`,
  [`401`]: `User unauthorized`,
};

export {
  AppRoute,
  Cities,
  ClassNames,
  MONTHS,
  RatingStarsCount,
  ScreenType,
  SortType,
  ServerMessage,
};
