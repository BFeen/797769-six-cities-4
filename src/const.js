const Offers = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal view Princengracht`,
 `Nice, cozy, warm big bed apartment`,
];

const Cities = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`,
];

const HousingTypes = [
  `Apartment`,
  `Room`,
  `House`,
  `Hotel`,
];

export default {Offers, Cities, HousingTypes};
