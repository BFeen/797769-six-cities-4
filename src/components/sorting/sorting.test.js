import React from "react";
import renderer from "react-test-renderer";
import Sorting from "./sorting.jsx";
import {SortType} from "../../common/const.js";


describe(`Sorting component snapshot testing`, () => {
  it(`Sorting rendering correctly`, () => {
    const tree = renderer.create(
        <Sorting
          isOpened={false}
          currentSortType={SortType.POPULAR}
          onSelectSortType={() => {}}
          onMenuClick={() => {}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Sorting with open menu rendering correctly`, () => {
    const tree = renderer.create(
        <Sorting
          isOpened={true}
          currentSortType={SortType.POPULAR}
          onSelectSortType={() => {}}
          onMenuClick={() => {}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
