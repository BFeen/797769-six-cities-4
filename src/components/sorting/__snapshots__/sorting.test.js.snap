// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting component snapshot testing Sorting renderinng correctly 1`] = `
<select
  className="places__sorting-type"
  defaultValue="popular"
  id="places-sorting"
  onChange={[Function]}
>
  <option
    className="places__option"
    value="popular"
  >
    Popular
  </option>
  <option
    className="places__option"
    value="to-high"
  >
    Price: low to high
  </option>
  <option
    className="places__option"
    value="to-low"
  >
    Price: high to low
  </option>
  <option
    className="places__option"
    value="top-rated"
  >
    Top rated first
  </option>
</select>
`;
